### PR TITLE
White dwarf meta knowledge check fix

### PR DESCRIPTION
**Pull request by grungussuss**
## About The Pull Request
There is two values for round threat, the shown value and the real value, sometimes the shown threat can be 0, while the real threat is higher, which in turn does not trigger the green alert on getting the classified report and does not make a special announcement, "Thanks to the tireless efforts of our security and intelligence divisions, there are currently no credible threats to [station_name()]. All station construction projects have been authorized. Have a secure shift!", nor does it enable all station construction projects, so we should disable being able to fake white dwarf on classified report.
## Why It's Good For The Game
Knowing when your classified report is fake is meta knowledge and shouldn't be a thing.
## Changelog
:cl: grungussuss
fix: fixed getting a fake white dwarf report when the shift isn't extended mode, which lead to meta knowledge being used.
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/dynamic/dynamic.dm [hunk 1]
```diff
@@ -387,9 +387,14 @@ SUBSYSTEM_DEF(dynamic)
...
-			advisory_string += "Advisory Level: <b>White Dwarf</b></center><BR>"
-			advisory_string += "Your sector's advisory level is White Dwarf. Our surveillance has ruled out any and all potential threats known in our database, eliminating most risks to our assets in the Spinward Sector. We advise a lower level of security, alongside distributing resources on potential profit."
+			if(round(threat_level) == 0)
+				advisory_string += "Advisory Level: <b>White Dwarf</b></center><BR>"
+				advisory_string += "Your sector's advisory level is White Dwarf. Our surveillance has ruled out any and all potential threats known in our database, eliminating most risks to our assets in the Spinward Sector. We advise a lower level of security, alongside distributing resources on potential profit."
+			else
+				advisory_string += "Advisory Level: <b>Green Star</b></center><BR>"
+				advisory_string += "Your sector's advisory level is Green Star. Surveillance information shows no credible threats to Nanotrasen assets within the Spinward Sector at this time. As always, the Department of Intelligence advises maintaining vigilance against potential threats, regardless of a lack of known threats."
```
Comment (MrMelbert) [inline code/controllers/subsystem/dynamic/dynamic.dm:397]: Don't copy paste the whole green alert text a second time, move this block out
```dm
if(round(shown_threat) == 0 && round(threat_level) == 0)
    return White dwarf
    
switch(round(shown_threat))
    if(0 to 19)
        string = Green Star
```

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/dynamic/dynamic.dm [hunk 2]
```diff
@@ -387,10 +387,11 @@ SUBSYSTEM_DEF(dynamic)
 		return advisory_string
 
 	switch(round(shown_threat))
-		if(0)
+		//a white dwarf shift leads to a green security alert on report and special announcement, this prevents a meta check if the alert report is fake or not.
+		if(0 && round(threat_level) == 0)
```
Comment (MrMelbert) [inline code/controllers/subsystem/dynamic/dynamic.dm:391]: This isn't how switches work, it needs to be moved out of the switch

Comment (Sadboysuss) [reply code/controllers/subsystem/dynamic/dynamic.dm]: done

File: code/controllers/subsystem/dynamic/dynamic.dm [hunk 3]
```diff
@@ -385,12 +385,14 @@ SUBSYSTEM_DEF(dynamic)
 		advisory_string += "Advisory Level: <b>Pulsar Star</b></center><BR>"
 		advisory_string += "Your sector's advisory level is Pulsar Star. A large, unknown electromagnetic field has stormed through nearby surveillance equipment, causing major data loss. Partial data was recovered and showed no credible threats to Nanotrasen assets within the Spinward Sector; however, the Department of Intelligence advises maintaining high alert against potential threats due to the lack of complete data."
 		return advisory_string
+	//a white dwarf shift leads to a green security alert on report and special announcement, this prevents a meta check if the alert report is fake or not.
+	if(0 && round(threat_level) == 0)
```
Comment (Jacquerel) [inline code/controllers/subsystem/dynamic/dynamic.dm:389]: Now you are just checking `if(0)` which is equivalent to `if(FALSE)`

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Pull request merged**